tests/Http/HttpRequestTest.php

```diff
@@ -122,7 +122,7 @@ public function testFullUrlMethod()
         $this->assertEquals('https://foo.com', $request->fullUrl());
 
         $request = Request::create('https://foo.com', 'GET');
-        $this->assertEquals('https://foo.com?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
+        $this->assertEquals('https://foo.com/?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
```

Comment:
An actual explanation on how that is the expected behavior would be awesome. It would take the same amount of text to write. I do not have the issue because I just dropped even using the method, but that is a bug or otherwise its intended use is actively misinterpreted by people, which would make at least for a note in docs or in the method doc block itself. Just to not push people into unnecessary debugging and head banging sessions.